test/lit/passes/merge-blocks-eh.wast

```diff
@@ -0,0 +1,214 @@
+  ;; CHECK-NEXT: )
+  (func $drop-block-try_catch_all_i32
+    ;; Send a tag value + exnref. We don't handle this yet TODO
+    (tuple.drop 2
+      (block $catch (result i32 exnref)
+        (try_table (catch_ref $i32 $catch)
+          (call $import)
+          (unreachable)
+        )
+      )
+    )
+  )
+
+  ;; CHECK:      (func $drop-block-try_catch_multi_partial (type $0)
+  ;; CHECK-NEXT:  (tuple.drop 2
+  ;; CHECK-NEXT:   (block $outer (type $1) (result i32 exnref)
+  ;; CHECK-NEXT:    (block $inner
+  ;; CHECK-NEXT:     (try_table (catch_ref $i32 $outer) (catch_all $inner)
+  ;; CHECK-NEXT:      (call $import)
+  ;; CHECK-NEXT:      (unreachable)
+  ;; CHECK-NEXT:     )
+  ;; CHECK-NEXT:    )
+  ;; CHECK-NEXT:    (unreachable)
+  ;; CHECK-NEXT:   )
+  ;; CHECK-NEXT:  )
+  ;; CHECK-NEXT: )
+  (func $drop-block-try_catch_multi_partial
+    ;; Two possible tags can be thrown and caught, and we can optimize one.
+    (tuple.drop 2
+      (block $outer (result i32 exnref)
+        (drop
+          (block $inner (result exnref)
+            (try_table (catch_ref $i32 $outer) (catch_all_ref $inner)
+              (call $import)
+              (unreachable)
+            )
+          )
+        )
+        (unreachable)
+      )
+    )
+  )
+
+  ;; CHECK:      (func $drop-block_try_catch_multi-fail (type $0)
+  ;; CHECK-NEXT:  (drop
+  ;; CHECK-NEXT:   (block $catch (result exnref)
+  ;; CHECK-NEXT:    (try_table (catch $exnref $catch) (catch_all_ref $catch)
+  ;; CHECK-NEXT:     (call $import)
+  ;; CHECK-NEXT:     (unreachable)
+  ;; CHECK-NEXT:    )
+  ;; CHECK-NEXT:   )
+  ;; CHECK-NEXT:  )
+  ;; CHECK-NEXT: )
+  (func $drop-block_try_catch_multi-fail
+    (drop
+      ;; This block is sent an exnref in two ways: once as the contents of a tag
+      ;; and once as the ref of a catch_all_ref. We can remove the latter but
+      ;; not the former, and they go to the same block, so we cannot optimize.
+      (block $catch (result exnref)
+        (try_table (catch $exnref $catch) (catch_all_ref $catch)
+          (call $import)
+          (unreachable)
+        )
+      )
+    )
+  )
+
+  ;; CHECK:      (func $drop-block-try_catch_all (type $0)
+  ;; CHECK-NEXT:  (block $catch
+  ;; CHECK-NEXT:   (try_table (catch_all $catch)
+  ;; CHECK-NEXT:    (call $import)
+  ;; CHECK-NEXT:    (unreachable)
+  ;; CHECK-NEXT:   )
+  ;; CHECK-NEXT:  )
+  ;; CHECK-NEXT: )
+  (func $drop-block-try_catch_all
+    ;; Without _ref, there is nothing to optimize (and we should not error).
+    (block $catch
+      (try_table (catch_all $catch)
+        (call $import)
+        (unreachable)
+      )
+    )
+  )
+
+  ;; CHECK:      (func $drop-block-try_catch (type $0)
+  ;; CHECK-NEXT:  (block $catch
+  ;; CHECK-NEXT:   (try_table (catch $empty $catch)
+  ;; CHECK-NEXT:    (call $import)
+  ;; CHECK-NEXT:    (unreachable)
+  ;; CHECK-NEXT:   )
+  ;; CHECK-NEXT:  )
+  ;; CHECK-NEXT: )
+  (func $drop-block-try_catch
+    ;; As above, without _all.
+    (block $catch
```

Comment:
Here too

Reply:
I improved the comment here but I'm not sure I understand your concern - in what way is it similar? (I assume you mean it is similar from "here too") The previous issue was about tags, but "tag" is not mentioned in this comment.

Reply:
Oh, i see, this is similar to the next comment, not the previous one...

Reply:
I meant this one https://github.com/WebAssembly/binaryen/pull/6996/files#r1796307595
(Isn't this the previous one?)

meaning both of them didn't have a `drop` anyway.

Reply:
Hmm, it shows up in the reverse order on the github UI, for me... strange. I guess it might emit comments on the main page in comment order and not code order.

Reply:
I was looking the comments in the code view.. But now I see that Github main page UI shows them in the swapped order. Why..? Not sure about how they decide the order (it's not the order I wrote them because I wouldn't have written "Here too" before the previous one...)... Anyway sorry for the confusion 😅